transactionoutbox-core/src/main/java/com/gruelbox/transactionoutbox/DefaultPersistor.java

```diff
@@ -294,7 +294,12 @@ public List<TransactionOutboxEntry> selectBatch(Transaction tx, int batchSize, I
                 + " WHERE groupId = t.groupId "
                 + "AND processed = false "
                 + "AND createdAt <= t.createdAt) < ? "
-                + "AND blocked = false "
```

Comment:
Should we really replace the `blocked = false` condition here or is the new condition rather an additional thing to consider? I.e., can we we ensure that entries **after** a blocked entry are never blocked themselves?

Reply:
Resolved: Entries after blocked ones are only skipped but never marked as blocked themselves since they aren't even necessarily blocked.